### PR TITLE
 gh-1 Return basic Order and OrderItem details to GetAllOrders REST API

### PR DESCRIPTION
**Pull request by Andrii77711**

### CODE REVIEW

File: src/main/java/com/to/ordermanagementservice/dto/OrderDetails.java [hunk 1]
```diff
@@ -1,10 +1,21 @@
...
+
+import java.time.OffsetDateTime;
+import java.util.List;
+
+public class OrderDetails {//todo reterne all list of order and user ()
 
     private Integer orderId;
 
-    private Integer customerId;
+    private Integer userId;// todo put castomer id not user
```
Comment (taras-oglabyak) [inline src/main/java/com/to/ordermanagementservice/dto/OrderDetails.java]: According to the https://github.com/Andrii77711/order-management-service/issues/1 it should be called `customerId`

File: src/main/java/com/to/ordermanagementservice/dto/OrderItemDetails.java [hunk 2]
```diff
@@ -10,6 +11,10 @@ public class OrderItemDetails {
 
     private Integer quantity;
 
+    private OffsetDateTime updateAt;
+
+    private OffsetDateTime createdAt;
+
     private BigDecimal price;
 
     public Integer getOrderItemId() {
```
Comment (taras-oglabyak) [inline src/main/java/com/to/ordermanagementservice/dto/OrderItemDetails.java]: should be called `id`

File: src/main/java/com/to/ordermanagementservice/entity/Order.java [hunk 3]
```diff
@@ -10,6 +16,18 @@ public class Order {
 
     private OffsetDateTime updatedAt;
 
+    private List<OrderItem> orderItems;
```
Comment (taras-oglabyak) [inline src/main/java/com/to/ordermanagementservice/entity/Order.java]: entities should not be link directly in our project

File: src/main/java/com/to/ordermanagementservice/repository/impl/InMemoryOrderRepository.java [hunk 4]
```diff
@@ -1,22 +1,25 @@
...
+import java.time.LocalDate;
 import java.time.OffsetDateTime;
 import java.util.List;
 
 @Repository
 public class InMemoryOrderRepository implements OrderRepository {
 
     private final List<Order> orders;
-
+    private final OrderItemRepository orderItemRepository;
```
Comment (taras-oglabyak) [inline src/main/java/com/to/ordermanagementservice/repository/impl/InMemoryOrderRepository.java]: see the comment above regarding entities linking. Please, inject both Repositories in Order Service and do entity->dto conversion there

File: src/main/resources/application.yaml [hunk 5]
```diff
@@ -3,4 +3,4 @@ spring:
     name: order-management-service
 
 server:
-  port: 18080
+  port: 18081
```
Comment (taras-oglabyak) [inline src/main/resources/application.yaml]: why did you change the port?

File: src/main/java/com/to/ordermanagementservice/web/controller/OrdersApiController.java [hunk 6]
```diff
@@ -11,7 +11,7 @@
 import java.util.List;
 
 @RestController
-@RequestMapping("/api/v1")
+@RequestMapping("/api/v1")//
```
Comment (taras-oglabyak) [inline src/main/java/com/to/ordermanagementservice/web/controller/OrdersApiController.java]: Is this comment needed?

File: src/main/java/com/to/ordermanagementservice/repository/impl/InMemoryOrderRepository.java [hunk 7]
```diff
@@ -1,22 +1,25 @@
 package com.to.ordermanagementservice.repository.impl;
 
 import com.to.ordermanagementservice.entity.Order;
+import com.to.ordermanagementservice.repository.OrderItemRepository;
 import com.to.ordermanagementservice.repository.OrderRepository;
 import org.springframework.stereotype.Repository;
 
+import java.time.LocalDate;
```
Comment (taras-oglabyak) [inline src/main/java/com/to/ordermanagementservice/repository/impl/InMemoryOrderRepository.java]: is this import used?

**Review by taras-oglabyak — CHANGES_REQUESTED** [review]
Please check all comments in this PR and remove unnecessary commetns from your code

File: src/main/java/com/to/ordermanagementservice/dto/OrderItemDetails.java [hunk 8]
```diff
@@ -1,15 +1,22 @@
...
 import java.math.BigDecimal;
+import java.time.OffsetDateTime;
 
 public class OrderItemDetails {
 
     private Integer orderItemId;
 
     private Integer productId;
 
+    private Integer id;
```
Comment (taras-oglabyak) [inline src/main/java/com/to/ordermanagementservice/dto/OrderItemDetails.java]: why do you nee both `orderItemId` and `id` in this class?

File: src/main/java/com/to/ordermanagementservice/entity/Order.java [hunk 9]
```diff
@@ -1,3 +1,8 @@
+//1.todo write into class (change custimer (on userid)) replace orderitem id here
+//2.todo setter metod and getter
+//3.todo add userid here
+
+
```
Comment (taras-oglabyak) [inline src/main/java/com/to/ordermanagementservice/entity/Order.java]: is this needed?

File: src/main/java/com/to/ordermanagementservice/entity/OrderItem.java [hunk 10]
```diff
@@ -1,3 +1,5 @@
+//no dublicate userid in orderitem (completed)
+
```
Comment (taras-oglabyak) [inline src/main/java/com/to/ordermanagementservice/entity/OrderItem.java]: is this needed?

File: src/main/java/com/to/ordermanagementservice/entity/OrderItem.java [hunk 11]
```diff
@@ -6,20 +8,14 @@
...
-
-    private Integer quantity;//кол.
+    private Integer quantity;
     private BigDecimal price;
-
     private OffsetDateTime createdAt;
-
     private OffsetDateTime updatedAt;
 
+    // Getters и Setters
```
Comment (taras-oglabyak) [inline src/main/java/com/to/ordermanagementservice/entity/OrderItem.java]: is this needed?

File: src/main/java/com/to/ordermanagementservice/service/impl/SimpleOrderService.java [hunk 12]
```diff
@@ -34,7 +39,27 @@ public List<OrderDetails> getAllOrders() {
...
+        orderDetails.setUpdatedAt(order.getUpdatedAt());
+        orderDetails.setCreatedAt(order.getCreatedAt());
         return orderDetails;
     }
+    private List<OrderItemDetails> collectOrderItemDetails (Integer orderId){
+        List<OrderItem> orderItems = orderItemRepository.getOrderItemsByOrderId(orderId);
+        List<OrderItemDetails> result = new ArrayList<>();
+        for (OrderItem  orderItem: orderItems){
+            OrderItemDetails orderItemDetails = new OrderItemDetails();
+            orderItemDetails.setOrderItemId(orderItem.getOrderId());
```
Comment (taras-oglabyak) [inline src/main/java/com/to/ordermanagementservice/service/impl/SimpleOrderService.java]: `orderItemId` is not an `orderId`. please check it

**Review by taras-oglabyak — CHANGES_REQUESTED** [review]

File: src/main/java/com/to/ordermanagementservice/dto/OrderItemDetails.java (hunk 8 above)
Comment (Andrii77711) [reply src/main/java/com/to/ordermanagementservice/dto/OrderItemDetails.java]: on task write 
Extend the GetAllOrders REST API to return all orders in the system:

Order entity
id
customerId
createdAt
updatedAt
orderItems[]
id
productId
quantity
price
createdAt
updatedAt

Comment (taras-oglabyak) [reply src/main/java/com/to/ordermanagementservice/dto/OrderItemDetails.java]: the "orderItemId" is not in the list

File: src/main/java/com/to/ordermanagementservice/dto/OrderDetails.java [hunk 13]
```diff
@@ -1,11 +1,20 @@
 package com.to.ordermanagementservice.dto;
 
+import java.time.OffsetDateTime;
+import java.util.List;
+
 public class OrderDetails {
 
     private Integer orderId;
```
Comment (taras-oglabyak) [inline src/main/java/com/to/ordermanagementservice/dto/OrderDetails.java]: it should be called `id`

**Review by taras-oglabyak — CHANGES_REQUESTED** [review]
